### PR TITLE
front-end User set up

### LINKED ISSUE #40: Build out front end styling
- [ ] Challenge List Page
    - [ ] Individual Challenge Page
    - [ ] ~~~ STYLE ~~~
    - [ ] Account preferences page/User submission history

### PR DESCRIPTION
**Pull request by KristyHSC**
set up the redux store on user
and a user component

close #40

### CODE REVIEW

File: client/store/user.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    case SET_USER:
+      return action.user;
+    default:
+      return state;
+  }
+};
+
+export const getUser = aUser => dispatch => {
+  console.log('loggingin store', aUser);
+  return axios.put('/api/users/login', aUser).then(user => dispatch(setUser(user)));
```
Comment (chem2016) [inline client/store/user.js]: update the route to axios.put('/api/users/auth/login', aUser) to be not mix with other put route

Comment (chem2016) [reply client/store/user.js]: Also add express.json() in the app.js in the backend

File: client/components/Login.js [hunk 2]
```diff
@@ -1,10 +1,61 @@
...
+
+  changeHandler(ev) {
+    this.setState({ [ev.target.name]: ev.target.value });
+  }
+
+  saveHandler(ev) {
+    ev.preventDefault();
+    this.props
+      .getUser(this.state)
+      .then(() => console.log('We have a uer logged in.'))
```
Comment (theo333) [inline client/components/Login.js]:
```suggestion
      .then(() => console.log('We have a user logged in.'))
```

File: server/routes/api/users.js [hunk 3]
```diff
@@ -40,4 +40,16 @@ router.delete('/:id', (req, res, next) => {
     .catch(next);
 });
 
+router.put('/auth/login', (req, res, next) => {
+  console.log('logging in route', req.body);
```
Comment (theo333) [inline server/routes/api/users.js]: Remove console.log()
```suggestion
  
```

**Pull request merged**